src/lib.rs

```diff
@@ -19,56 +19,69 @@ pub struct Args {
 
 pub fn configure(parsed_args: Args) -> Result<Config> {
     // Set defaults
-    let mut config = Config::default();
-    config
-        .set_default::<&str>("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))
+    let mut builder = Config::builder();
+    builder = builder
+        .set_default("cfg_file", concat!(env!("CARGO_PKG_NAME"), ".toml"))
         .context(ErrorKind::ConfigImmutable)?;
-    config
-        .set_default::<&str>("Log.file", "_CONSOLE_")
+    builder = builder
+        .set_default("Log.file", "_CONSOLE_")
         .context(ErrorKind::ConfigImmutable)?;
-    config
-        .set_default::<&str>("Web.address", "127.0.0.1")
+    builder = builder
+        .set_default("Web.address", "127.0.0.1")
         .context(ErrorKind::ConfigImmutable)?;
-    config
+    builder = builder
         .set_default("Web.port", 7878)
         .context(ErrorKind::ConfigImmutable)?;
 
     // Determine config file
-    if let Ok(config_file) = env::var("AARDWOLF_CONFIG") {
-        config
-            .set("cfg_file", config_file)
-            .context(ErrorKind::ConfigImmutable)?;
-    }
+    let config_file = env::var("AARDWOLF_CONFIG").unwrap();
+    builder = builder
+        .set_override("cfg_file", config_file.clone())
+        .context(ErrorKind::ConfigImmutable)?;
```

Comment:
I'm struggling a little to follow what you're asking for - I'm still a bit of a Rust novice, and I'm unfamiliar with the project structure.

Are you asking for `config_file` on line 37 to be populated from `parsed_args.config` instead of unwrapping the `env::var("AARDWOLF_CONFIG")` ?

Reply:
Sorry for the confusion, but yes, that is exactly what I was requesting.